explorer/src/Pos/Explorer/Socket/App.hs

```diff
@@ -174,9 +181,13 @@ periodicPollChanges connVar closed =
                             Just blocks -> return blocks
             let newBlunds = fromMaybe [] mNewBlunds
 
-            -- notify about blocks and blocks with offset
+            -- notify changes depending on new blocks
             unless (null newBlunds) $ do
+                -- 1. last page of blocks
                 notifyBlocksLastPageSubscribers
+                -- 2. last page of epochs
+                mSlotId :: Maybe SlotId <- lift $ getCurrentSlot @ctx
```

Comment:
Remove type signature ` :: Maybe SlotId`? It would also remove the extra import for the `SlotId`.